draft-ietf-quic-transport.md

```diff
@@ -3309,103 +3288,70 @@ ACK Delay:
 
-~~~
- 0                   1                   2                   3
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                      First ACK Block (i)                    ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                             Gap (i)                         ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                    Additional ACK Block (i)                 ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                             Gap (i)                         ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                    Additional ACK Block (i)                 ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-                               ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                             Gap (i)                         ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                    Additional ACK Block (i)                 ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-~~~
-{: #ack-block-format title="ACK Block Section"}
+```
+ack_type_i = decode_varint() & 3
+```
 
-Each ACK Block acknowledges a contiguous range of packets by indicating the
-number of acknowledged packets that precede the largest packet number in that
-block.  A value of zero indicates that only the largest packet number is
-acknowledged.  Larger ACK Block values indicate a larger range, with
-corresponding lower values for the smallest packet number in the range.  Thus,
-given a largest packet number for the ACK, the smallest value is determined by
-the formula:
+The four types of ACK Block are:
 
-~~~
-   smallest = largest - ack_block
-~~~
+Gap (0):
 
-The range of packets that are acknowledged by the ACK block include the range
-from the smallest packet number to the largest, inclusive.
+: A gap describes a range of packets that are not acknowledged.  This does not
+  mean that these packets were not received, but that this ACK frame does not
+  include an acknowledgment for those packets.
 
-The largest value for the First ACK Block is determined by the Largest
-Acknowledged field; the largest for Additional ACK Blocks is determined by
-cumulatively subtracting the size of all preceding ACK Blocks and Gaps.
+Unmarked (1):
 
-Each Gap indicates a range of packets that are not being acknowledged.  The
-number of packets in the gap is one higher than the encoded value of the Gap
-Field.
+: Unmarked packets are those packets that are acknowledged, but for which the
+  ECN marking is either not present, ECT(1), or unknown.
 
-The value of the Gap field establishes the largest packet number value for the
-ACK block that follows the gap using the following formula:
+ECT(0) Marked (2):
```

Comment:
I wonder if this type should be called just ECT Marked (2). And then make it clear that only packet with an expected ECT codepoint is to be reported. The default behavior is that only ECT(0) SHALL be reported using this type. Future extensions may change which code points that are reported using this type. That way extensions, such as a L4S congestion control extension in the future can define that ECT(1) is the one to be reported using this type after the end-point has agreed on that.